File: backend/src/main/java/com/zeptopluse/auth/AuthDtos.java
package com.zeptopluse.auth;

import jakarta.validation.constraints.*;
import java.time.LocalDateTime;

public final class AuthDtos {
    private AuthDtos() {}

    public record RegisterRequest(
            @NotBlank @Size(max = 120) String name,
            @NotBlank @Email @Size(max = 160) String email,
            @Size(max = 20) String phone,
            @NotBlank @Size(min = 8, max = 72) String password) {}

    public record LoginRequest(
            @NotBlank @Email @Size(max = 160) String email,
            @NotBlank String password) {}

    public record RefreshRequest(@NotBlank String refreshToken) {}

    public record AuthUser(Long id, String name, String phone, String email, String role) {}

    public record AuthResponse(String accessToken, String refreshToken, long expiresInSeconds, AuthUser user) {}
}

File: app/src/main/java/com/daily/nexamartpartner/features/auth/data/model/AuthDtos.kt
package com.daily.nexamartpartner.features.auth.data.model

import com.squareup.moshi.Json

data class AuthUserDto(
    @field:Json(name = "id") val id: Long?,
    @field:Json(name = "name") val name: String?,
    @field:Json(name = "phone") val phone: String?,
    @field:Json(name = "email") val email: String?,
    @field:Json(name = "role") val role: String?
)

data class LoginResponseDto(
    @field:Json(name = "accessToken") val accessToken: String?,
    @field:Json(name = "refreshToken") val refreshToken: String?,
    @field:Json(name = "user") val user: AuthUserDto?
)
